silq/tools/plot_tools.py

```diff
@@ -821,7 +833,15 @@ def update_plot(self, initialize=False):
             initialize: Method called during initialization.
         """
         for k, name in enumerate(self.parameter.names):
-            result = self.parameter.results[name]
+            if self.buffer_length > 1:
+                assert self.parameter.trace_pulse.enabled is False, \
+                    'Trace pulse must be turned off for running average.'
```

Comment:
why is this?

Reply:
Because if `self.parameter.trace_pulse.enabled is True`, then `self.parameter.results[name]` contains also data from trace_pulse and it complicates implementation of this "averaging thing". I agree this is not a neat solution, can try to improve